ui/src/lib.rs

```diff
@@ -1021,205 +1026,232 @@ impl Ui for Tui {
                 last_index = draw_memory.inner_call_index;
             }
             // Grab interrupt
-            match rx.recv()? {
-                // Key press
-                Interrupt::KeyPressed(event) => match event.code {
-                    // Exit
-                    KeyCode::Char('q') => {
-                        disable_raw_mode()?;
-                        execute!(
-                            self.terminal.backend_mut(),
-                            LeaveAlternateScreen,
-                            DisableMouseCapture
-                        )?;
-                        return Ok(TUIExitReason::CharExit)
-                    }
-                    // Move down
-                    KeyCode::Char('j') | KeyCode::Down => {
-                        // Grab number of times to do it
-                        for _ in 0..Tui::buffer_as_number(&self.key_buffer, 1) {
-                            if event.modifiers.contains(KeyModifiers::CONTROL) {
-                                let max_mem = (debug_call[draw_memory.inner_call_index].1
-                                    [self.current_step]
-                                    .memory
-                                    .len() /
-                                    32)
-                                .saturating_sub(1);
-                                if draw_memory.current_mem_startline < max_mem {
-                                    draw_memory.current_mem_startline += 1;
+
+            let receiver = rx.recv()?;
+
+            if let Some(c) = receiver.char_press() {
+                if self.key_buffer.ends_with('\'') {
+                    if let Some((caller, pc)) = self.breakpoints.get(&c) {
+                        for (i, (_caller, debug_steps, _)) in debug_call.iter().enumerate() {
+                            if _caller == caller {
+                                if let Some(step) =
+                                    debug_steps.iter().position(|step| step.pc == *pc)
+                                {
+                                    draw_memory.inner_call_index = i;
+                                    self.current_step = step;
+                                    break
```

Comment:
what's the change here?
I'd like to see docs for this

Reply:
Had to scope it to avoid repeating the part about the matching of the pressed char